pages/chaoss/visualizations/contrib_importance_pie.py

```diff
@@ -0,0 +1,262 @@
+                                    ], 
+                                    className = "me-2",
+                                ),
+                                 dbc.Col(
+                                    [dbc.Button(
+                                        "About Graph",
+                                        id=f"popover-target-{PAGE}-{VIZ_ID}",
+                                        color="secondary",
+                                        size="sm",
+                                    ), ],
+                                    width="auto",
+                                    style={"paddingTop": ".5em"},
+                                ),
+                            ], 
+                            align="center",
+                        ), 
+                    ]
+                ),
+            ]
+        )
+    ],
+)
+
+# callback for graph info popover
+@callback(
+    Output(f"popover-{PAGE}-{VIZ_ID}", "is_open"),
+    [Input(f"popover-target-{PAGE}-{VIZ_ID}", "n_clicks")],
+    [State(f"popover-{PAGE}-{VIZ_ID}", "is_open")],
+)
+def toggle_popover(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+# callback for contrib-importance graph
+@callback(
+    Output(f"{PAGE}-{VIZ_ID}", "figure"),
+    Output(f"check-alert-{PAGE}-{VIZ_ID}", "is_open"),
+    [
+        Input("repo-choices", "data"),
+        Input(f"action-type-{PAGE}-{VIZ_ID}", "value"),
+        Input(f"patterns-{PAGE}-{VIZ_ID}", "value"),
+        Input(f"top-k-contributors-{PAGE}-{VIZ_ID}", "value"),
+
+    ],
+    background=True,
+)
+
+def create_top_10_cntrbs_graph(repolist, action_type, patterns, top_k):
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=ctq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=ctq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.warning(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.warning(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph, False
+    
+    # checks if there is a contribution of a specfic action type in repo set
+    if not df["Action"].str.contains(action_type).any():
+        return dash.no_update, True
+    
+    # function for all data pre processing
+    df = process_data(df, action_type, patterns, top_k)
+
+    fig = create_figure(df, action_type)
+
+    logging.warning(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig, False
+
+def process_data(df: pd.DataFrame, action_type, patterns, top_k):
+
+    # subset the df such that it only contains rows where the Action column value is the action type
+    df = df[df["Action"].str.contains(action_type)]
+
+    # option to filter out potential bots
+    if patterns:
+        # remove rows where Login column value contains the substring 'bot'
+        df = df[df['login'].str.contains(str(patterns)[1:-1])==False]
```

Comment:
Use ~ not ==False and use the following logic: https://www.skytowner.com/explore/getting_rows_where_column_value_contains_any_substring_in_a_list_in_pandas_dataframe

Reply:
Included `na=False` clause in mask variable since there are some nan/empty login column values.

Reply:
Have you specifically check if using ~ works for nan and empty values? Every check ive done shows that it would

Reply:
When I just do ~, it returns `TypeError: bad operand type for unary -: 'NoneType' ` so I referenced this post: https://stackoverflow.com/questions/52297740/typeerror-bad-operand-type-for-unary-float